library/std/src/sys/xous/locks/rwlock.rs

```diff
@@ -0,0 +1,67 @@
+use crate::cell::Cell;
+
+pub struct RwLock {
+    // This platform has no threads, so we can use a Cell here.
+    mode: Cell<isize>,
```

Comment:
You've implemented threads, so this is is probably a mistake

Reply:
That's a very good point, thank you. I've replaced it with a version that uses `AtomicISize` and now spins while it waits for locks, calling `do_yield()` in between.

Ideally we'd delegate this to an external server to coordinate in case of contention, but I think that can wait for now.